Deprecate 'auth' attribute and create an approved list of attributes

mbryzek opened the pull request:
- Deprecate 'auth' attribute
- [RFC](https://docs.google.com/document/d/14CRL6j28yKq-cMAQrxM7wbC1uGNqTa5i8BP9zHnAHXs/edit)
- [Partner API](https://github.com/flowcommerce/api-partner/pull/53)
- [API](https://github.com/flowcommerce/api/pull/2068)
- [API Internal](https://github.com/flowcommerce/api-internal/pull/2216)

benwaffle commented:
What do you think about throwing an error on any unrecognized attribute?

Otherwise lgtm

mbryzek commented:
> What do you think about throwing an error on any unrecognized attribute?

Updated to have a known list of attributes - agree this is better (it found the now deprecated object-reference attribute which I've also removed)
